diff -uNr zlib-1.3/configure zlib-1.3.mod/configure
--- zlib-1.3/configure	2023-08-18 11:45:36.000000000 +0300
+++ zlib-1.3.mod/configure	2023-11-29 01:51:17.924387123 +0200
@@ -232,7 +232,9 @@
         *sparc*)
             LDFLAGS="${LDFLAGS} -Wl,--no-warn-rwx-segments" ;;
         esac
-        LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1,--version-script,${SRCDIR}zlib.map"} ;;
+        shared_ext='.so'
+        SHAREDLIB='libz_1.so'
+        LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz_1.so,--version-script,${SRCDIR}zlib.map"} ;;
   *BSD | *bsd* | DragonFly)
         LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1,--version-script,${SRCDIR}zlib.map"}
         LDCONFIG="ldconfig -m" ;;
@@ -439,7 +441,7 @@
 extern int getchar();
 int hello() {return getchar();}
 EOF
-if test $shared -eq 1; then
+if false; then
   echo Checking for shared library support... | tee -a configure.log
   # we must test in two steps (cc then ld), required at least on SunOS 4.x
   if try $CC -c $SFLAGS $test.c &&
